internal/bminventory/inventory.go

```diff
@@ -4868,8 +4883,13 @@ func (b *bareMetalInventory) RegisterInfraEnvInternal(ctx context.Context, kubeK
 			infraEnv.ProxyHash = infraEnvProxyHash
 		}
 
+		var mirroredRegistries []string
+		if mirrorRegistryConfiguration != nil {
+			mirroredRegistries = extractMirroredRegistriesFromConfig(log, mirrorRegistryConfiguration)
```

Comment:
What happens if extraction fails?

Reply:
As agreed below let's put the comment on extractMirroredRegistriesFromConfig, that way we should only need to do it once.
I think it's reasonable for the reader to look at the function and see the comment.

Reply:
Added the comment below, thanks!